unit-tests/py/rspy/device_hub.py

```diff
@@ -0,0 +1,188 @@
+        :param reset: When true, the hub will be reset as part of the connection process
+        """
+        pass
+
+    @abstractmethod
+    def is_connected(self):
+        """
+        :return: True if the hub is connected, False otherwise
+        """
+        pass
+
+    @abstractmethod
+    def disconnect(self):
+        """
+        Disconnect from the hub
+        """
+        pass
+
+    @abstractmethod
+    def all_ports(self):
+        """
+        :return: a list of all possible ports, even if currently unoccupied or disabled
+        """
+        pass
+
+    @abstractmethod
+    def ports(self):
+        """
+        :return: a list of all ports currently enabled
+        """
+        pass
+
+    @abstractmethod
+    def is_port_enabled( self, port ):
+        """
+        query if the input port number is enabled through the hub
+        :param port: port number;
+        :return: True if the hub enabled this port, False otherwise
+        """
+        pass
+
+    @abstractmethod
+    def port_state( self, port ):
+        pass
+
+    @abstractmethod
+    def enable_ports( self,  ports = None, disable_other_ports = False, sleep_on_change = 0 ):
+        """
+        Set enable state to provided ports
+        :param ports: List of port numbers; if not provided, enable all ports
+        :param disable_other_ports: if True, the ports not in the list will be disabled
+        :param sleep_on_change: Number of seconds to sleep if any change is made
+        :return: True if no errors found, False otherwise
+        """
+        pass
+
+    @abstractmethod
+    def disable_ports( self, ports = None, sleep_on_change = 0 ):
+        """
+        :param ports: List of port numbers; if not provided, disable all ports
+        :param sleep_on_change: Number of seconds to sleep if any change is made
+        :return: True if no errors found, False otherwise
+        """
+        pass
+
+    @abstractmethod
+    def _get_port_by_loc(self, usb_location, hubs):
+        """
+        """
+        pass
+
+    def recycle_ports(self, portlist = None, timeout = 2 ):
+        """
+        Disable and enable a port
+        :param timeout: how long to wait before re-enabling
+        :return: True if everything OK, False otherwise
+        """
+        if portlist is None:
+            portlist = self.ports()
+        #
+        result = self.disable_ports( portlist )
+        #
+        import time
+        time.sleep( timeout )
+        #
+        result = self.enable_ports( portlist ) and result
+        #
+        return result
+
+    def get_port_by_location(self, physical_port, usb_location, hubs):
+        port = None
+        try:
+            port = self._get_port_by_loc(usb_location, hubs)
+        except Exception as e:
+            log.e('Failed to get device port:', e)
+            log.d('    physical port is', physical_port)
+            log.d('    USB location is', usb_location)
+
+        return usb_location, port
```

Comment:
If it's called `get_port_by_location` and it accepts the `usb_location`. then why are we returning it?